### PR TITLE
[MRG] Add support for adding dig points from montage.

### PR DESCRIPTION
**Pull request by jaeilepp**
Allows reading the digitization points from a saved ``DigMontage``.

### CODE REVIEW

**Comment by jaeilepp** [comment]
Ready for review.

File: mne/gui/_file_traits.py [hunk 1]
```diff
@@ -285,14 +287,40 @@ def _get_n_omitted(self):
     @cached_property
     def _get_inst(self):
         if self.file:
-            info = read_info(self.file, verbose=False)
-            if info['dig'] is None:
+            try:
+                info = read_info(self.file, verbose=False)
+            except ValueError:  # try reading dig_montage
```
Comment (christianbrodbeck) [inline mne/gui/_file_traits.py]: I don't like silencing errors that might be real. 

Could the modification be in `read_info()`, which could create an info dict when given a montage file?

**Comment by agramfort** [comment]
+1 for merge when CIs are happy and what's new is updated.

File: mne/gui/_file_traits.py (hunk 1 above)
Comment (jaeilepp) [inline mne/gui/_file_traits.py]: How about just logging the error message before looking for DigMontage? I don't like the idea of changing the behavior of the public function.

Comment (christianbrodbeck) [inline mne/gui/_file_traits.py]: That will clutter the log with an error message in the other case where there is no error to be concerned about. I think returning an info dict with the montage would be a logical extension of `read_info` (only change a case that was not functional previously), but if you don't want to change `read_info`, can you read the header of the file to determine what kind of file it is and treat it accordingly?

**Comment by agramfort** [comment]
@christianbrodbeck merge when you're happy

**Comment by jaeilepp** [comment]
Now it checks the file header.

**Comment by agramfort** [comment]
thx @jaeilepp

**Comment by jaeilepp** [comment]
Should I update the SPM dataset again and use the dig montage instead of the raw file?

**Comment by agramfort** [comment]
it would be cleaner indeed. thx

**Pull request merged**